africanus/calibration/tests/test_utils.py

```diff
@@ -5,19 +5,14 @@
 from __future__ import print_function
 
 import numpy as np
+from numpy.random import normal
+import pytest
 from numpy.testing import assert_array_almost_equal
 from africanus.dft import im_to_vis
 from africanus.averaging.support import unique_time
-from africanus.calibration.utils import residual_vis, correct_vis
+from africanus.calibration.utils import residual_vis, correct_vis, corrupt_vis
 from africanus.rime.predict import predict_vis
 
-n_time = 32
-n_chan = 16
-n_ant = 7
-n_bl = n_ant*(n_ant-1)//2
-n_row = n_bl*n_time
-n_cor = 2
-
 
 def give_lm(n_dir):
```

Comment:
This is a https://en.wikipedia.org/wiki/Factory_method_pattern

```suggestion
def lm_factory(n_dir):
```